libsolidity/ast/ASTPrinter.cpp

```diff
@@ -45,7 +45,8 @@ ASTPrinter::ASTPrinter(
 void ASTPrinter::print(ostream& _stream)
 {
 	m_ostream = &_stream;
-	m_ast->accept(*this);
+	if (m_ast != nullptr)
```

Comment:
This cannot happen - the ast is passed to this component by reference, so it should be non-null by convention.

Reply:
Yeah, I am not sure why I did this. It is reverted in a0865a891 after testing. But in doing this I see that this PR is very lacking in tests overall and will be addressing that too.